src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -25,23 +27,32 @@
     /**
      * Ignore the styles from the generic folder.
      */
-    private static final String NULL_STYLE = "schematics";
+    private static final String NULL_STYLE = "";
 
     /**
      * Ignore the styles from the miner folder.
      */
     private static final String MINER_STYLE = "miner";
 
-    private static final String                    SCHEMATIC_EXTENSION   = ".nbt";
-    private static final String                    SCHEMATICS_ASSET_PATH = "/assets/minecolonies/schematics/";
+    private static final String                    SCHEMATIC_EXTENSION        = ".nbt";
+    private static final String                    SCHEMATICS_ASSET_PATH      = "/assets/minecolonies/schematics/";
+    public  static final String                    SCHEMATICS_HUTS            = "huts";
+    public  static final String                    SCHEMATICS_DECORATIONS     = "decorations";
     //Hut, Levels
     @NotNull
-    private static final Map<String, Integer>      hutLevelsMap          = new HashMap<>();
+    private static       Map<String, Integer>      hutLevelsMap          = new HashMap<>();
     //Hut, Styles
     private static       Map<String, List<String>> hutStyleMap           = new HashMap<>();
     //Decoration, Style
     private static       Map<String, List<String>> decorationStyleMap    = new HashMap<>();
 
+    /* md5 hash for the schematics
+     * format is:
+     * huts/stone/builder1.nbt -> hash
+     * decorations/decoration/Well.nbt -> hash
+     */
+    private static       Map<String, String>       md5Map                = new HashMap<>();
```

Comment:
Why don't we enter the folder(style) and fileName into the hash? This would make it quite unique to avoid confusion.
If we want to prevent Hash colisions we shouldn't use md5 anyway. But Md5 is, for this use case, more than enough imo.